js/app.js

```diff
@@ -0,0 +1,64 @@
+
+// Declaro variable para solicitar el número de tarjeta desde un prompt
+var card = prompt('Ingrese número de Tarjeta de Credito a Validar:');
+// Evalua si card esta vacio o está ingresando letras
+if (card === '' || isNaN(card)) {
+// de ser true, brinda un mensaje
+  alert('Esta ingresando campo vacio o letras, Ingrese los datos');
+} else { // caso contrario llama a la funcion isValidCard
+  document.write(isValidCard(card));
+}
+// Funcion para validar el # de una tarjeta de crédito
+function isValidCard(num) {
+// Declaro un arreglo que contendra todos los elementos del número ingresado
+  var arr = [];
+  // Declaro variable que almacenara la validez de la tarjeta
+  var message = '';
+  // recorro el texto ingresado
+  for (var i = 0;i < num.length;i++) {
+  // Declaro variable que almacena el # de la tarjeta ingresada
+    var element = parseInt(num.charAt(i));
+    // agrego el elemento al array creado
+    arr.push(element);
+  }
+  // Declaro un nuevo array que contenga los números en orden invertido
+  var newArr = arr.reverse();
+  // console.log(newArr);
+  // Recorro el arreglo posicion por posicion
+  for (var j = 1;j <= newArr.length;j++) {
+  // evalua si la posicion es par
+    if (j % 2 === 0) {
+      // Declaro variable que almacena el doble del numero
+      var doublePair = newArr[j - 1] * 2;
+      // Evaluo si el doble de ese número es mayor o igual a 10
+      if (doublePair >= 10) {
+      // Separamos los digitos del resultado
+        var digitOne = parseInt(doublePair / 10);
+        var digitTwo = doublePair % 10;
+        // sumamos los  digitos del resultado
+        var sumDigit = digitOne + digitTwo;
+        // Reemplaza en cada posicion par los nuevos numeros
+        newArr[j - 1] = sumDigit;
+      } else {
+        // reemplazamos en cada posicion por el doble del elemento
+        newArr [j - 1] = doublePair;
+      }
+    }
+  }
+  // Declaro variable que almacenara la suma de todos los elementos del arreglo nuevo
+  var sumElement = 0;
+  for (var index = 0;index < newArr.length;index++) {
+    sumElement += newArr[index];
+  }
```

Comment:
Para calcular la suma de los elementos estás utilizando un nuevo ciclo `for`, esto significa que para revisar si un número de tarjeta es válido recorres el arreglo 16 x 2 = 32 posiciones, una optimización sería realizar la suma en el primer ciclo `for` directamente.

Si declaras la variable `sumElement` arriba del *primer* ciclo `for`, después en la última línea del ciclo puedes escribir:
`sumElement += newArr [j - 1];` y borrar el *segundo* ciclo `for`.